hack/common.sh

```diff
@@ -9,3 +9,22 @@ trap_add() {
   # shellcheck disable=SC2064 # Quotes are required here to properly expand when adding the new trap.
   trap "${hdls}${hdls:+;}${1:?Handler required}" "${sig}"
 }
+
+function prepend_generated_by_header() {
+  dst_file="$1"
+  generated_by="$2"
+
+  tmp_dir=$(mktemp -d)
+  trap 'rm -rf $tmp_dir' RETURN
+
+  generated_by_header_file="${tmp_dir}/header.txt"
+  printf "#=================================================================\n\
+#                 DO NOT EDIT THIS FILE\n\
+#  IT HAS BEEN GENERATED BY %s\n\
+#=================================================================\n\
+" "$generated_by" >"$generated_by_header_file"
+
+  tmp_dst_file="${tmp_dir}/dst.txt"
+  cat "$generated_by_header_file" "$dst_file" >"$tmp_dst_file"
```

Comment:
there are some simple ways to do it using sed -i. pls check if this reduces the code https://superuser.com/questions/246837/how-do-i-add-text-to-the-beginning-of-a-file-in-bash

Reply:
I actually did this at first (and ended up reading that same link!) :smile: 

IMO, that method more difficult to understand, and easier to break.

I think our goal is code that is easier to understand, and to maintain, even at the expense of length, or performance (e.g. this PR is relatively "slow," because it writes temp files, instead of doing in-memory replacement).